riscv/src/small_field/bootloader.rs

```diff
@@ -1 +1,100 @@
+use std::marker::PhantomData;
 
+use powdr_number::KnownField;
+use powdr_number::{FieldElement, LargeInt};
+
+use crate::continuations::bootloader::BootloaderImpl;
+
+use crate::continuations::bootloader::WORDS_PER_PAGE;
+
+pub fn split_word<F: FieldElement>(v: u32) -> (F, F) {
+    ((v & 0xffff).into(), (v >> 16).into())
+}
+
+pub fn split_fe<F: FieldElement>(v: &F) -> [F; 2] {
```

Comment:
So this takes 1 FE and splits into 2 16-bit limbs still in FE?

Reply:
yes: this is quite confusing, but is like that because of the poseidon machine which takes F instead of half-words.
So a `Hash` for the small field a `[F;8]`, the full `F`. But that eventually needs to become [F;16] (i.e., 16 half words), that our machine uses.
This similar for Goldilocks: poseidon takes `[F;4]`, full `F`, but we then need to split into `[F;8]` (i.e., 8 words).

Reply:
yea I see